### PR TITLE
[SMTChecker] Support bound function calls

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on #5307

### CODE REVIEW

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -457,6 +457,14 @@ void SMTChecker::inlineFunctionCall(FunctionCall const& _funCall)
 	else if (_funDef && _funDef->isImplemented())
 	{
 		vector<smt::Expression> funArgs;
+		auto const& funType = dynamic_cast<FunctionType const*>(_funCall.expression().annotation().type.get());
```
Comment (axic) [inline libsolidity/formal/SMTChecker.cpp]: There's a `_calledExpr` variable up there for part of this expression.

File: libsolidity/formal/SMTChecker.cpp [hunk 2]
```diff
@@ -457,6 +457,14 @@ void SMTChecker::inlineFunctionCall(FunctionCall const& _funCall)
...
+		solAssert(funType, "");
+		if (funType->bound())
+		{
+			auto const& boundFunction = dynamic_cast<MemberAccess const*>(&_funCall.expression());
```
Comment (axic) [inline libsolidity/formal/SMTChecker.cpp]: Same here.

**Comment by leonardoalt** [comment]
Updated also the rest of the function removing the `_` prefix for non-parameters.

**Comment by axic** [comment]
> also the rest of the function removing the _ prefix for non-parameters.

Would have preferred as a separate commit to improve clarity about what is the actual change for bound functions.

**Comment by axic** [comment]
@leonardoalt actually please do that and merge it after

**Comment by leonardoalt** [comment]
@axic Done, can you approve?

**Review by axic — APPROVED** [review]

**Pull request merged**